data/pigui/game.lua

```diff
@@ -0,0 +1,512 @@
+		ui.addLine(a, b, color, thickness)
+	end
+	local size = 2
+	pitchline(3, size * 2, ui.theme.colors.reticuleCircle, 1)
+	pitchline(2.25, size, ui.theme.colors.reticuleCircle, 1)
+	pitchline(3.75, size, ui.theme.colors.reticuleCircle, 1)
+	pitchline(1.5, size * 2, ui.theme.colors.reticuleCircle, 1)
+	pitchline(4.5, size * 2, ui.theme.colors.reticuleCircle, 1)
+	local xpitch = (pitch_degrees + 90) / 180
+	local xpitch_h = 4.5 - xpitch * 3
+	pitchline(xpitch_h, size * 3, ui.theme.colors.navigationalElements, 2)
+end
+
+local function displayReticuleHorizon(center, roll_degrees)
+	local offset = 30
+	local width = 10
+	local height_hrs = 0.1
+	local hrs = roll_degrees / 360 * 12 + 3
+	-- left hook
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, hrs),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset - width, hrs),
+						 ui.theme.colors.navigationalElements, 1)
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, hrs),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset, hrs + height_hrs),
+						 ui.theme.colors.navigationalElements, 1)
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, -3),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset + width/2, -3),
+						 ui.theme.colors.navigationalElements, 1)
+	-- right hook
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, hrs + 6),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset - width, hrs + 6),
+						 ui.theme.colors.navigationalElements, 1)
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, hrs + 6),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset, hrs + 6 - height_hrs),
+						 ui.theme.colors.navigationalElements, 1)
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius - offset, 3),
+						 ui.pointOnClock(center, reticuleCircleRadius - offset + width/2, 3),
+						 ui.theme.colors.navigationalElements, 1)
+
+
+
+end
+
+local function displayReticuleCompass(center, heading)
+	local relevant = {}
+	local directions = { [0] = lc.COMPASS_N, [45] = lc.COMPASS_NE, [90] = lc.COMPASS_E, [135] = lc.COMPASS_SE, [180] = lc.COMPASS_S, [225] = lc.COMPASS_SW, [270] = lc.COMPASS_W, [315] = lc.COMPASS_NW }
+	local function cl(x)
+		if x < 0 then
+			return cl(x + 360)
+		elseif x >= 360 then
+			return cl(x - 360)
+		else
+			return x
+		end
+	end
+	local left = math.floor(heading - 45)
+	local right = left + 90
+	local d = left
+
+	local function stroke(d, p, n, height, thickness)
+		if d % n == 0 then
+			local a = ui.pointOnClock(center, reticuleCircleRadius, 2.8 * p - 1.4)
+			local b = ui.pointOnClock(center, reticuleCircleRadius + height, 2.8 * p - 1.4)
+			ui.addLine(a, b, ui.theme.colors.reticuleCircle, thickness)
+		end
+	end
+
+	ui.addLine(ui.pointOnClock(center, reticuleCircleRadius, 0),
+						 ui.pointOnClock(center, reticuleCircleRadius - 3, 0),
+						 ui.theme.colors.reticuleCircle, 1)
+
+	while true do
```

Comment:
Now that's just a shame. Why not a for loop ? Much less error-prone and much more readable, too.